ipn/ipnlocal/c2n.go

```diff
@@ -115,17 +115,24 @@ func (b *LocalBackend) handleC2NUpdate(w http.ResponseWriter, r *http.Request) {
 	// TODO(bradfitz): add some sort of semaphore that prevents two concurrent
 	// updates, or if one happened in the past 5 minutes, or something.
 
-	var res tailcfg.C2NUpdateResponse
-	res.Enabled = envknob.AllowsRemoteUpdate()
-	res.Supported = runtime.GOOS == "windows" || (runtime.GOOS == "linux" && distro.Get() == distro.Debian)
-
-	switch r.Method {
-	case "GET", "POST":
-	default:
+	// GET returns the current status, and POST actually begins an update.
+	if r.Method != "GET" && r.Method != "POST" {
 		http.Error(w, "bad method", http.StatusMethodNotAllowed)
 		return
 	}
 
+	// If NewUpdater does not return an error, we can update the installation.
+	// Exception: When version.IsMacSysExt returns true, we don't support that
+	// yet. TODO(cpalmer, #6995): Implement it.
+	//
+	// Note that we create the Updater solely to check for errors; we do not
+	// invoke it here. For this purpose, it is ok to pass it a zero Arguments.
+	_, err := clientupdate.NewUpdater(clientupdate.Arguments{})
+	res := tailcfg.C2NUpdateResponse {
+		Enabled: envknob.AllowsRemoteUpdate(),
```

Comment:
Confirming we can't set envknobs remotely? I remember something like that, but my memory could be faulty

Reply:
There's https://github.com/tailscale/tailscale/blob/main/control/controlknobs/controlknobs.go but it's not automatically wired to envknobs (which is good).
We're adding a `tailscale set` flag in https://github.com/tailscale/tailscale/pull/8861 for user opt-in.